### PR TITLE
Cleanup mixer/pkg/api and get coverage to 100%

### PR DESCRIPTION
**Pull request by geeknoid**

### CODE REVIEW

File: mixer/pkg/api/grpcServer.go [hunk 1]
```diff
@@ -81,139 +80,107 @@ func NewGRPCServer(dispatcher runtime.Dispatcher, gp *pool.GoroutinePool) mixerp
 
 // Check is the entry point for the external Check method
 func (s *grpcServer) Check(legacyCtx legacyContext.Context, req *mixerpb.CheckRequest) (*mixerpb.CheckResponse, error) {
-	// TODO: this code doesn't distinguish between RPC failures when communicating with adapters and
```
Comment (ldemailly) [inline mixer/pkg/api/grpcServer.go]: is this todo actually done in some earlier PR ?
same for the next one

File: mixer/pkg/api/grpcServer.go [hunk 2]
```diff
@@ -81,139 +80,107 @@ func NewGRPCServer(dispatcher runtime.Dispatcher, gp *pool.GoroutinePool) mixerp
...
-	} else {
-		log.Errora("Check returned with error: ", status.String(out))
-	}
-
 	if status.IsOK(resp.Precondition.Status) && len(req.Quotas) > 0 {
 		resp.Quotas = make(map[string]mixerpb.CheckResponse_QuotaResult, len(req.Quotas))
-		var qr *mixerpb.CheckResponse_QuotaResult
 
-		// TODO: should dispatch this loop in parallel
-		// WARNING: if this is dispatched in parallel, then we need to do
```
Comment (ldemailly) [inline mixer/pkg/api/grpcServer.go]: seems like an interesting todo and warning?

File: mixer/pkg/api/grpcServer_test.go [hunk 3]
```diff
@@ -288,7 +347,13 @@ func TestReport(t *testing.T) {
...
 	}
 
-	request := mixerpb.ReportRequest{Attributes: []mixerpb.CompressedAttributes{{}}}
+	request := mixerpb.ReportRequest{Attributes: []mixerpb.CompressedAttributes{}}
+	_, err = ts.client.Report(context.Background(), &request)
+	if err != nil {
+		t.Errorf("Expected success, got error: %v", err)
+	}
+
+	request = mixerpb.ReportRequest{Attributes: []mixerpb.CompressedAttributes{{}}}
```
Comment (ozevren) [inline mixer/pkg/api/grpcServer_test.go]: Seems like line 350 & 356 are same?

Comment (geeknoid) [reply mixer/pkg/api/grpcServer_test.go]: Subtelly different, the second case has an extra set of braces...

File: mixer/pkg/api/grpcServer.go (hunk 1 above)
Comment (ldemailly) [reply mixer/pkg/api/grpcServer.go]: @geeknoid ^ ?

**Pull request merged**